oauthproxy.go

```diff
@@ -943,15 +945,18 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 		return nil, ErrNeedsLogin
 	}
 
-	invalidEmail := session != nil && session.Email != "" && !p.Validator(session.Email)
-	invalidGroups := session != nil && !p.validateGroups(session.Groups)
+	invalidEmail := session.Email != "" && !p.Validator(session.Email)
+	authorized, err := p.provider.Authorize(req.Context(), session)
+	if err != nil {
+		logger.Errorf("Error with authorization: %v", err)
+	}
 
-	if invalidEmail || invalidGroups {
-		logger.Printf(session.Email, req, logger.AuthFailure, "Invalid authentication via session: removing session %s", session)
+	if invalidEmail || !authorized {
+		logger.PrintAuthf(session.Email, req, logger.AuthFailure, "Invalid authentication via session: removing session %s", session)
 		// Invalid session, clear it
 		err := p.ClearSessionCookie(rw, req)
 		if err != nil {
-			logger.Printf("Error clearing session cookie: %v", err)
+			logger.Errorf("Error clearing session cookie: %v", err)
 		}
 		return nil, ErrNeedsLogin
```

Comment:
I m seeing too many ERR requests, We need to throw an error if the user is not part allowedgroups isn it ?
I changed this line which fixed too many redirects issue 
return nil, fmt.Errorf("Permission denied ,Invalid account or not part of autherized group")

Reply:
Good point - that makes sense.  At this point of the code, a user has already authenticated, but they aren't authorized `if invalidEmail || !authorized`

I'll flip this to a different error.  Thanks for the help!

I take it this issue exists now (at least with email authorization/validation) since I didn't touch this logic in my refactor?

Reply:
Yes..it exists now . Can you fix this as well. Thanks

Reply:
https://github.com/oauth2-proxy/oauth2-proxy/pull/797/commits/c34ec47ffb9e49cb6f49e0985166b05e2bc8b72b

Reply:
Perfect .Thanks .